component/common/src/main/java/org/wso2/carbon/identity/smsotp/common/util/Utils.java

```diff
@@ -157,6 +165,18 @@ public static SMSOTPClientException handleClientException(Constants.ErrorMessage
         return new SMSOTPClientException(error.getCode(), error.getMessage(), description);
     }
 
+    public static SMSOTPClientException handleClientException(Constants.ErrorMessage error, String data,
+                                                              int remainingFailedAttempts) {
+
+        String description;
+        if (StringUtils.isNotBlank(data)) {
+            description = String.format(error.getDescription(), data, remainingFailedAttempts);
+        } else {
+            description = error.getDescription();
+        }
+        return new SMSOTPClientException(error.getCode(), error.getMessage(), description);
+    }
+
```

Comment:
We should be able to call the following method from the above method so that we can get rid of duplications.
```
public static SMSOTPClientException handleClientException(Constants.ErrorMessage error, String data,
                                                              int remainingFailedAttempts, Throwable e) {
```

Reply:
Resolved with 69b280575c654b56917dae1fdf53ea71618ef29a